api/event/event.go

```diff
@@ -0,0 +1,240 @@
+package event
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"mime"
+	"net/http"
+	"strings"
+
+	"github.com/fnproject/fn/api/common"
+)
+
+const DefaultCloudEventVersion = "0.1"
+
+//
+//type BodyType int
+//
+//// TODO Handle body conversion/derivation
+//const (
+//	BodyTypeNone   = BodyType(iota)
+//	BodyTypeJSON   = BodyType(iota)
+//	BodyTypeString = BodyType(iota)
+//	BodyTypeBinary = BodyType(iota)
+//)
+
+// TODO :  Ideally this would be able to pass an arbitrary byte stream in its body (i.e. not be subject to JSON limitations)   but it's not now
+// Currently this only accepts valid JSON bodies, or non-json content that is valid UTF-8
+// TODO: Would really prefer this to be an interface with stronger correctness by guarantee
```

Comment:
i'm not sure we need to support binary _in_ a json. spec release 0.2 maybe clarifies this some more (i.e. base64).